src/validation/block.cairo

```diff
@@ -70,5 +73,10 @@ pub fn compute_and_validate_tx_data(
     validate_transactions?;
     validate_block_weight(total_weight)?;
 
-    Result::Ok((total_fee, merkle_root(ref txids), merkle_root(ref wtxids)))
+    let mut wtxid_merkle_root = Default::default();
+    if block_height >= SEGWIT_BLOCK {
```

Comment:
You can do `let wtxid_root = if block_height >= SEGWIT_BLOCK {} else {}` to avoid mut